packages/expo/cli/start/doctor/ngrok/ExternalModule.ts

```diff
@@ -0,0 +1,194 @@
+    shouldGloballyInstall,
+  }: InstallPromptOptions = {}): Promise<IModule> {
+    const packageName = [this.pkg.name, this.pkg.versionRange].join('@');
+    if (!autoInstall) {
+      // Delay the prompt so it doesn't conflict with other dev tool logs
+      await delayAsync(100);
+    }
+    const answer =
+      autoInstall ||
+      (shouldPrompt &&
+        (await confirmAsync({
+          message: this.promptMessage(packageName),
+          initial: true,
+        })));
+    if (answer) {
+      Log.log(`Installing ${packageName}...`);
+
+      // Always use npm for global installs
+      const packageManager = shouldGloballyInstall
+        ? new PackageManager.NpmPackageManager({
+            cwd: this.projectRoot,
+            log: Log.log,
+            silent: !EXPO_DEBUG,
+          })
+        : PackageManager.createForProject(this.projectRoot, {
+            silent: !EXPO_DEBUG,
+          });
+
+      try {
+        if (shouldGloballyInstall) {
+          await packageManager.addGlobalAsync(packageName);
+        } else {
+          await packageManager.addDevAsync(packageName);
+        }
+        Log.log(`Installed ${packageName}`);
+      } catch (e) {
+        e.message = `Failed to install ${packageName} ${
+          shouldGloballyInstall ? 'globally' : 'locally'
+        }: ${e.message}`;
+        throw e;
+      }
+      return await this.resolveAsync({ shouldPrompt: false });
+    }
+
+    throw new CommandError(
+      'EXTERNAL_MODULE_AVAILABILITY',
+      `Please install ${packageName} and try again`
+    );
+  }
+
+  /** Get the module. */
+  get(): IModule | null {
+    try {
+      return this.getVersioned();
+    } catch {
+      return null;
+    }
+  }
+
+  /** Get the module, throws if the module is not versioned correctly. */
+  getVersioned(): IModule | null {
+    if (!this.instance) {
+      this.instance = this._resolveModule(true) ?? this._resolveModule(false);
+    }
+
+    return this.instance ?? null;
+  }
+
+  /** Exposed for testing. */
+  _require(moduleId: string): any {
+    return require(moduleId);
+  }
+
+  /** Resolve a copy that's installed in the project. Exposed for testing. */
+  _resolveLocal(moduleId: string): string {
+    return resolveFrom(this.projectRoot, moduleId);
+  }
+
+  /** Resolve a copy that's installed globally. Exposed for testing. */
+  _resolveGlobal(moduleId: string): string {
+    return requireGlobal.resolve(moduleId);
+  }
+
+  /** Resolve the module and verify the version. Exposed for testing. */
+  _resolveModule(isLocal: boolean): IModule | null {
+    const resolver = isLocal ? this._resolveLocal : this._resolveGlobal;
+    try {
+      const packageJsonPath = resolver(`${this.pkg.name}/package.json`);
+      const packageJson = this._require(packageJsonPath);
+      if (packageJson) {
+        if (semver.satisfies(packageJson.version, this.pkg.versionRange)) {
+          const modulePath = resolver(this.pkg.name);
+          return this._require(modulePath);
```

Comment:
May want to assert that this is not null. If a package exports null or undefined (module.exports = null) it will break a lot of logic in this class, so better to break cleanly.